src/utils/comet.ts

```diff
@@ -0,0 +1,111 @@
+import { Network, TxOptions, i128, invokeOperation } from '@blend-capital/blend-sdk';
+import { Contract, nativeToScVal, scValToBigInt, xdr } from 'stellar-sdk';
+
+export class CometClient {
+  comet: Contract;
+
+  constructor(address: string) {
+    this.comet = new Contract(address);
+  }
+  /**
+ * 
+    token_in: Address,
+    token_amount_in: i128,
+    min_pool_amount_out: i128,
+    user: Address,
+ */
+  public async depositTokenInGetLPOut(
+    source: string,
+    sign: (txXdr: string) => Promise<string>,
+    network: Network,
+    depositTokenAddress: string,
+    depositTokenAmount: bigint,
+    minLPTokenAmount: bigint,
+    user: string,
+    txOptions: TxOptions
+  ) {
+    const invokeArgs = {
+      method: 'dep_tokn_amt_in_get_lp_tokns_out',
+      args: [
+        nativeToScVal(depositTokenAddress, { type: 'address' }),
+        nativeToScVal(depositTokenAmount, { type: 'i128' }),
+        nativeToScVal(minLPTokenAmount, { type: 'i128' }),
+        nativeToScVal(user, { type: 'address' }),
+      ],
+    };
+    const operation = this.comet.call(invokeArgs.method, ...invokeArgs.args);
+    return await invokeOperation<i128>(
+      source,
+      sign,
+      network,
+      txOptions,
+      (value: string | xdr.ScVal | undefined): i128 | undefined => {
+        if (value == undefined) {
+          return undefined;
+        }
+        console.log({ value });
+        const scVal = scValToBigInt(xdr.ScVal.fromXDR(value as string, 'base64'));
+        console.log({ scVal });
+        return scVal;
+      },
+      operation
```

Comment:
Going to need to convert to xdr for change that occurred in blend sdk ie operation.toXDR('base64')